### PR TITLE
CM-367_edit_returns_for_errors_from_api_endpoints

### PR DESCRIPTION
**Pull request by y-himanen**
- incorporated Sean's edits to readme and docstrings
- edited error returns as per strategy agreed with Sean
- edited HTTP status codes for accuracy
- edited store_subscription_data function to stop app crashing when invalid session id is used
- ran local migration so all local tables are up-to-date
- NOTE - a discussion should be had between front and backend regarding whether the foreign key constraint should be dropped from the signup table if the user should be able to subscribe without a session id

### CODE REVIEW

**Comment by y-himanen** [comment]
i would be grateful if both @seanmajorpayne and @NickCallaghan could test this to see if anything was missed. please test the api endpoints wildly and creatively for any possible scenario where a user may decide to do something we don't want/expect them to do. i tried to do that myself, but may of course have missed something (again!)

**Review by seanmajorpayne — APPROVED** [review]
Thank you for incorporating my changes 💯 
Everything ran great in Postman for me. I'll leave the more exhaustive testing to Nick since it sounds like he already has an idea of what he's looking for.

File: knowledge_graph/routes.py [hunk 1]
```diff
@@ -53,24 +66,23 @@ def subscribe():
         session_id = request_body["sessionId"]
         response = store_subscription_data(session_id, email)
         return response
-    except Exception as e:
-        print(e)
-        return {"error": "Invalid Request"}, 400
+    except:
+        return {"message": "Invalid request"}, 400
```
[the remark this answers is not in the document]
Comment (y-himanen) [reply knowledge_graph/routes.py:70]: Handled in function called by endpoint

File: knowledge_graph/routes.py [hunk 2]
```diff
@@ -53,24 +66,23 @@ def subscribe():
...
+    Returns the list of available schwartz personal value questions that can be
+    presented to the user.
     """
     try:
         file = os.path.join(os.getcwd(), "json_files", "schwartz_questions.json")
         with open(file) as json_file:
             data = load(json_file)
     except FileNotFoundError:
-        return make_response("Schwartz Questions not Found"), 400
+        return {"message": "Schwartz questions not found"}, 404
```
[the remark this answers is not in the document]
Comment (y-himanen) [reply knowledge_graph/routes.py:85]: 404 is for file not found

**Review by NickCallaghan — APPROVED** [review]

File: knowledge_graph/routes.py [hunk 3]
```diff
@@ -106,9 +108,8 @@ def receive_user_scores() -> Tuple[Response, int]:
     try:
         parameter = request.json
     # todo: handle exceptions properly here
-    except Exception as ex:
-        print(ex)
-        return make_response({"error": "Invalid User Response"}), 400
+    except:
+        return {"message": "Invalid user response"}, 400
```
[the remark this answers is not in the document]
Comment (y-himanen) [reply knowledge_graph/routes.py:112]: this one has been written in a succeed or fail way. i think it requires a new ticket/more thought if we want more scenarios covered.

File: knowledge_graph/routes.py [hunk 4]
```diff
@@ -247,7 +237,7 @@ def get_personal_values():
         return jsonify(response), 200
 
     else:
-        return make_response("Invalid Session ID - Internal Server Error"), 400
+        return {"message": "Invalid session ID"}, 500
 
 
 @app.route("/get_actions", methods=["GET"])
```
[the remark this answers is not in the document]
Comment (y-himanen) [reply knowledge_graph/routes.py:243]: can't quite tell which endpoint you're referring to?

**Pull request merged**